src/cloud/digitalocean/provisioner.ts

```diff
@@ -67,9 +68,21 @@ class Provisioner {
     return this.doOAuth_().then((oauthObj: any) => {
       this.state_.oauth = oauthObj;
       return this.getSshKey_(name);
-      // Get SSH keys
     }).then((keys: KeyPair) => {
+       // Get SSH keys
       this.state_.ssh = keys;
+      return this.getDroplet_(name).catch((e: Error) => {
```

Comment:
Why don't you do something like:
```typescript
return this.getDroplet_(name).then((unused:any) => {
  // Droplet exists so raise an error.
  return Promise.reject({
    'errcode': 'VM_AE',
    'message': 'Droplet ' + name + ' already exists'
  });
}, (e:Error) => {
  return this.setupDigitalOcean_(name, region, image, size);
});
```
In fact, I don't see how it works right now.
 +        });